File: linked_lists/linked_list_cycle.py
class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if not head or not head.next:
            return False

        fast, slow = head, head

        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
            if fast is slow:
                    return True 

        return False

File: linked_lists/linked_list_cycle.rb
def hasCycle(head)
  return false if head.nil? || head.next.nil?

  fast, slow = head, head


	while !fast.nil? && !fast.next.nil?
        return true if fast == slow
        slow = slow.next
        fast = fast.next.next 
	end

	false
end